### DISCUSSION TITLE
Sparse Gaussian Processes

### DISCUSSION THREAD

**Discussion by GHU2021** [cornellius-gp/gpytorch#2020] [Q&A] [answered]
Hi,

In Pyro, there is a model called SparseGPRegression to implement Sparse Gaussian Processes.
Is there any similar model in GPyTorch for Sparse Gaussian Processes?
If not, how can we use current models for this purpose.
I want to implement Sparse Gaussian Processes, Sparse Multioutput Gaussian Processes, and Sparse Multitask Gaussian Processes.

I appreciate you help.

**Comment by gpleiss** [comment, marked as answer]
See [the SGPR (sparse GP regression) example](https://docs.gpytorch.ai/en/stable/examples/02_Scalable_Exact_GPs/SGPR_Regression_CUDA.html).

It should be compatible with the [multioutput and multitask GPs](https://docs.gpytorch.ai/en/stable/examples/03_Multitask_Exact_GPs/index.html#multi-output-vector-valued-functions). Just wrap your `covar_module` with `InducingPointKernel` (as in the SGPR example).

**Reply by GHU2021** [reply]
Dear Dr. Pleiss,

Thank you for quick response and helpful information.
